### ISSUE TITLE
Bidders with usersync issues

### ISSUE THREAD

**Issue by bretg** [prebid/prebid-server#2832]
We've done an runtime analysis of the usersyncs for quite a few PBS bidders and found several of them have issues. 

We're asking the following companies to take a look at the corresponding section below:

- [x] pubmatic - @pm-nilesh-chate 
- [x] colossus - @HuddledMasses
- [x] adkernel - @ckbo3hrk
- [x] criteo - @afewcc 
- [x] ucfunnel - @guscarreon
- [x] marsmedia - @vladi-mmg 
- [ ] aja - @sambaiz 
- [x] amx - @nickjacob 
- [x] lockerdome - @margsliu 
- [x] synacormedia - @ecammit 
- [x] connectad - @rtuschkany
- [x] conversant
- [x] smaato
- [x] 33across

## Pubmatic - the iframe script is not generating a call to PBS
```
https://ads.pubmatic.com/AdServer/js/user_sync.html?gdpr=0&gdpr_consent=&us_privacy=1---&predirect=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dpubmatic%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Db%26uid%3D
```

## Colossus - the redirect from https://sync.colossusssp.com/pbs.gif is not resolving the [UID] macro. 
```
/setuid?bidder=colossus&gdpr=0&gdpr_consent=&us_privacy=1---&gpp=&gpp_sid=&account=&f=i&uid=[UID]
```

## adkernel - returns HTTP 200 rather than 302
```
https://sync.adkernel.com/user-sync?t=image&gdpr=0&gdpr_consent=&us_privacy=1---&r=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dadkernel%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D%7BUID%7D
```

## Criteo iframe not producing /setuid to PBS
```
https://ssp-sync.criteo.com/user-sync/iframe?gdprapplies=0&gdpr=&ccpa=1---&redir=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dcriteo%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Db%26uid%3D%24%7BCRITEO_USER_ID%7D&profile=230
```
The HTML result has a return code of 200 and the body contains several <img> tags but none of them point to PBS.

Yes, we're running the latest version of PBS-Java (1.120) which contains the recent iframe change for your adapter.

## ucfunnel - many redirects, but none to PBS
```
https://sync.aralego.com/idsync?gdpr=0}&gdpr_consent=&usprivacy=1---&redirect=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Ducfunnel%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3DSspCookieUserId
```
redirects to
```
https://match.adsrvr.org/track/cmf/generic?ttd_pid=nwuslum&ttd_tpi=1&gdpr=0&gdpr_consent=
```
and then to
```
https://sync.aralego.com/idsync?ucf_nid=dsp-9A2296D7D389BBA3144262983D2B9AEB&ucf_user_id=f33ddccd-28a5-48f5-86ef-589ae32490af
```
and then to
```
https://x.bidswitch.net/sync?ssp=ucfunnel&user_id=20516ce9-aef2-3e5b-b898-e5259d38c8f3&gdpr=0&gdpr_consent=
```
and so on -- several more.  But never back to PBS.

## Marsmedia - call to PBS doesn't contain an actual UID
```
https://dmp.rtbsrv.com/dmp/profiles/cm?p_id=179&gdpr=0&gdpr_consent=&us_privacy=1---&redirect=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dmarsmedia%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D%24%7BUUID%7D
```
after several redirects with encoded bodies, the result is a call to Prebid Server that doesn't actually carry the uid:
```
https://prebid-server.rubiconproject.com/setuid?bidder=marsmedia&gdpr=0&gdpr_consent=&us_privacy=1---&gpp=&gpp_sid=&account=&f=i&uid=
```

## AJA - returns HTTP 200 instead of 302
```
https://ad.as.amanad.adtdp.com/v1/sync/ssp?ssp=4&gdpr=0&us_privacy=1---&redir=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Daja%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D%25s
```

## AMX - redirects to a different place that returns HTTP 204 rather than a redirect to Prebid Server
```
https://prebid.a-mo.net/cchain/0?gdpr=0&gdpr_consent=&us_privacy=1---&cb=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Damx%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D
```
Redirects to
```
https://ap.lijit.com/pixel?gdpr=1&gdpr_consent=&us_privacy=1---&redir=https%3A%2F%2Fprebid.a-mo.net%2Fcchain%2F1%2F458%3Fgdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26A%3D66c31a72-9e67-4e4e-b1ac-9d6821e9ed15%26bidder%3Dsovrn%26cbx%3DaHR0cHM6Ly9wcmViaWQtc2VydmVyLnJ1Ymljb25wcm9qZWN0LmNvbS9zZXR1aWQ_YmlkZGVyPWFteCZnZHByPTAmZ2Rwcl9jb25zZW50PSZ1c19wcml2YWN5PTEtLS0mZ3BwPSZncHBfc2lkPSZhY2NvdW50PSZmPWkmdWlkPQ%253D%253D%26uid%3D%24UID
```
which returns HTTP 204

## Synacormedia - iframe doesn't call PBS /setuid
```
https://ad-cdn.technoratimedia.com/html/usersync.html?cb=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dsynacormedia%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Db%26uid%3D%5BUSER_ID%5D
```
The HTML result has a return code of 200 and a minified body of code, but never results in calling the redirect defined in the `cb` parameter.

Is this possibly the result of your recent name change to imds?

## Connectad - iframe doesn't call PBS /setuid
```
https://cdn.connectad.io/connectmyusers.php?gdpr=0&consent=&us_privacy=1---&cb=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dconnectad%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Db%26uid%3
```
The HTML result has a return code of 200 and a minified body of code, but never results in calling the redirect defined in the `cb` parameter.

## Lockerdome - returns HTTP 200 rather than 302
```
https://lockerdome.com/usync/prebidserver?redirect=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dlockerdome%26gdpr%3D0%26gdpr_consent%3D%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D%7B%7Buid%7D%7D
```

## Conversant - returning 204 when syncing in GDPR scenario
```
https://prebid-match.dotomi.com/match/bounce/current?version=1&networkId=72582&rurl=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dconversant%26gdpr%3D1%26gdpr_consent%3DCPtGDMAPtGDMAAGABCENDECkAP_AAG_AAAYgI3td_H_fbX9j-_594ft0eY1f8Pq_ruAzDheNk-8FyJ_W_LwX_2EyNF36pq4KmR4Eu1LBIQNlHEHUDQmQYIkFqTHsak2MpzNKJqJEklMZO2dYGF9vmxFD-YKY705-9vfx2Du9_dv-z5T3r8Xd0AAQACAEAIACAAAAAAAAQAAAAAAAgIAAAAAAAAAAgAQAABAIJIgEGAgIAAACCAAEACAEIAAIABIAAAAAAgAAoAEACBQABKwCIgEAAFAIAIQAAQggIQIBAAABAEgAAAgBYIBAABAIAAQAAAAAACAAEBABYCAAAAgEgAABAACBAQRAAAQhAQEABBACwAAAAEEhgAACACAEAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAIAAAAAgLBIIgACAAFwAUABUADIAHAAPAAgABgADKAGgAagA8gCGAIoATAAngBVADeAHMAPQAfgBCACGAEcAJYATQApQBbgDDAGQAMsAbIA7wB7AD4gH2AfsA_wEAAIpARcBGACNAEcAJMASkAoIBTwCrgFzAMUAawA2gBuAEOgJEATKAnYBQ4CjwFIgKaAWKAtgBcgC8wGDAMNAZOAy4BnIDSAIhhABIAGwASAE0AKWAWcBNgCmwFuAMCDQGAAuACGAGQAMsAbMA-wD8AIAAQUAjABJgCngFXgLQAtIBrADqgIdARMAioBIgCdgFIgLkAZOAzkMACATYApsRAUAEMAMgAZYA2YB9gH4AQAAjABJgCngFXANYAdUBDoCRAE7AKRAXIAyMBk4DORAAIAEgCbBUA4ACgAQwAmABcAEcAMsAjgBV4C0ALSAtgBcgC8wGRgM5FAAQE2DIBoAQwAmACOAGWARwAq4BWwFogLYAXIAvMBkYDORgAEBNg6CmAAuACgAKgAZAA4ACAAF0AMAAygBoAGoAPAAfQBDAEUAJgATwAqgBcADEAGYAN4AcwA9AB-AEMAJYATQAowBSgCxAFvAMIAwwBkADKAGiANkAd4A9oB9gH6AP8AikBFgEYgI4AjoBJgCUgFBAKeAVcAsUBaAFpALmAXkAxQBtADcAHOAOoAfYBDoCKgEXgJEASoAnYBQ4CjwFNAKsAWKAtgBcAC5AF2gLzAYMAw0BjwDIwGTgMqAZcAzMBnIDRAGkAOLHgAQCMjgAoAJABHAJsAU2AtwhAyAAWABQADIALgAYgA1ACGAEwAKYAVQAuABiADMAG8APQAjgBSgCxAGEAMoAd4A-wB_gEUAI4ASkAoIBTwCrwFoAWkAuYBigDaAHOAOoAkQBKgCmgFWALFAWUAtEBbAC4AFyALtAZGAycBnIDRCIAIAjICYiAAIAZoCbCUC4ABAACwAKAAZAA4AB-AGAAYgA8ACIAEwAKoAXAAxABmAEMAI4AUYApQBbgDCAGUANkAd8A-wD8AI4AU8Aq8BaAFpALmAYoA3AB1AEOgImARUAi8BIgCjwFigLYAXaAvMBkYDJwGcgNIJgAQCMkgAYAzQE2ALcKQPAAFwAUABUADIAHAAQQAwADKAGgAagA8gCGAIoATAAngBSACqAGIAMwAcwA_ACGAFGAKUAWIAtwBhADIAGUANEAbIA74B9gH6ARYAjEBHAEdAJSAUEAq4BWwC5gF5ANoAbgA-wCHQETAIvASIAk4BOwCrAFigLYAXAAuQBdoC8wGGgMjAZOAy4BnIDSCoAEANpQAMACQAP4BBwE2AKbAW4A.YAAAAAAAAAAA%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D
```

## Smaato - returning 204 when syncing in GDPR scenario
```
https://s.ad.smaato.net/c/?adExInit=p&redir=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3Dsmaato%26gdpr%3D1%26gdpr_consent%3DCPtGDMAPtGDMAAGABCENDECkAP_AAG_AAAYgI3td_H_fbX9j-_594ft0eY1f8Pq_ruAzDheNk-8FyJ_W_LwX_2EyNF36pq4KmR4Eu1LBIQNlHEHUDQmQYIkFqTHsak2MpzNKJqJEklMZO2dYGF9vmxFD-YKY705-9vfx2Du9_dv-z5T3r8Xd0AAQACAEAIACAAAAAAAAQAAAAAAAgIAAAAAAAAAAgAQAABAIJIgEGAgIAAACCAAEACAEIAAIABIAAAAAAgAAoAEACBQABKwCIgEAAFAIAIQAAQggIQIBAAABAEgAAAgBYIBAABAIAAQAAAAAACAAEBABYCAAAAgEgAABAACBAQRAAAQhAQEABBACwAAAAEEhgAACACAEAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAIAAAAAgLBIIgACAAFwAUABUADIAHAAPAAgABgADKAGgAagA8gCGAIoATAAngBVADeAHMAPQAfgBCACGAEcAJYATQApQBbgDDAGQAMsAbIA7wB7AD4gH2AfsA_wEAAIpARcBGACNAEcAJMASkAoIBTwCrgFzAMUAawA2gBuAEOgJEATKAnYBQ4CjwFIgKaAWKAtgBcgC8wGDAMNAZOAy4BnIDSAIhhABIAGwASAE0AKWAWcBNgCmwFuAMCDQGAAuACGAGQAMsAbMA-wD8AIAAQUAjABJgCngFXgLQAtIBrADqgIdARMAioBIgCdgFIgLkAZOAzkMACATYApsRAUAEMAMgAZYA2YB9gH4AQAAjABJgCngFXANYAdUBDoCRAE7AKRAXIAyMBk4DORAAIAEgCbBUA4ACgAQwAmABcAEcAMsAjgBV4C0ALSAtgBcgC8wGRgM5FAAQE2DIBoAQwAmACOAGWARwAq4BWwFogLYAXIAvMBkYDORgAEBNg6CmAAuACgAKgAZAA4ACAAF0AMAAygBoAGoAPAAfQBDAEUAJgATwAqgBcADEAGYAN4AcwA9AB-AEMAJYATQAowBSgCxAFvAMIAwwBkADKAGiANkAd4A9oB9gH6AP8AikBFgEYgI4AjoBJgCUgFBAKeAVcAsUBaAFpALmAXkAxQBtADcAHOAOoAfYBDoCKgEXgJEASoAnYBQ4CjwFNAKsAWKAtgBcAC5AF2gLzAYMAw0BjwDIwGTgMqAZcAzMBnIDRAGkAOLHgAQCMjgAoAJABHAJsAU2AtwhAyAAWABQADIALgAYgA1ACGAEwAKYAVQAuABiADMAG8APQAjgBSgCxAGEAMoAd4A-wB_gEUAI4ASkAoIBTwCrwFoAWkAuYBigDaAHOAOoAkQBKgCmgFWALFAWUAtEBbAC4AFyALtAZGAycBnIDRCIAIAjICYiAAIAZoCbCUC4ABAACwAKAAZAA4AB-AGAAYgA8ACIAEwAKoAXAAxABmAEMAI4AUYApQBbgDCAGUANkAd8A-wD8AI4AU8Aq8BaAFpALmAYoA3AB1AEOgImARUAi8BIgCjwFigLYAXaAvMBkYDJwGcgNIJgAQCMkgAYAzQE2ALcKQPAAFwAUABUADIAHAAQQAwADKAGgAagA8gCGAIoATAAngBSACqAGIAMwAcwA_ACGAFGAKUAWIAtwBhADIAGUANEAbIA74B9gH6ARYAjEBHAEdAJSAUEAq4BWwC5gF5ANoAbgA-wCHQETAIvASIAk4BOwCrAFigLYAXAAuQBdoC8wGGgMjAZOAy4BnIDSCoAEANpQAMACQAP4BBwE2AKbAW4A.YAAAAAAAAAAA%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Di%26uid%3D%24UID
```

## 33Across - returning 204 when syncing in GDPR scenario
```
https://ssc-cms.33across.com/ps/?m=xch&rt=html&gdpr=1&gdpr_consent=CPtGDMAPtGDMAAGABCENDECkAP_AAG_AAAYgI3td_H_fbX9j-_594ft0eY1f8Pq_ruAzDheNk-8FyJ_W_LwX_2EyNF36pq4KmR4Eu1LBIQNlHEHUDQmQYIkFqTHsak2MpzNKJqJEklMZO2dYGF9vmxFD-YKY705-9vfx2Du9_dv-z5T3r8Xd0AAQACAEAIACAAAAAAAAQAAAAAAAgIAAAAAAAAAAgAQAABAIJIgEGAgIAAACCAAEACAEIAAIABIAAAAAAgAAoAEACBQABKwCIgEAAFAIAIQAAQggIQIBAAABAEgAAAgBYIBAABAIAAQAAAAAACAAEBABYCAAAAgEgAABAACBAQRAAAQhAQEABBACwAAAAEEhgAACACAEAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAIAAAAAgLBIIgACAAFwAUABUADIAHAAPAAgABgADKAGgAagA8gCGAIoATAAngBVADeAHMAPQAfgBCACGAEcAJYATQApQBbgDDAGQAMsAbIA7wB7AD4gH2AfsA_wEAAIpARcBGACNAEcAJMASkAoIBTwCrgFzAMUAawA2gBuAEOgJEATKAnYBQ4CjwFIgKaAWKAtgBcgC8wGDAMNAZOAy4BnIDSAIhhABIAGwASAE0AKWAWcBNgCmwFuAMCDQGAAuACGAGQAMsAbMA-wD8AIAAQUAjABJgCngFXgLQAtIBrADqgIdARMAioBIgCdgFIgLkAZOAzkMACATYApsRAUAEMAMgAZYA2YB9gH4AQAAjABJgCngFXANYAdUBDoCRAE7AKRAXIAyMBk4DORAAIAEgCbBUA4ACgAQwAmABcAEcAMsAjgBV4C0ALSAtgBcgC8wGRgM5FAAQE2DIBoAQwAmACOAGWARwAq4BWwFogLYAXIAvMBkYDORgAEBNg6CmAAuACgAKgAZAA4ACAAF0AMAAygBoAGoAPAAfQBDAEUAJgATwAqgBcADEAGYAN4AcwA9AB-AEMAJYATQAowBSgCxAFvAMIAwwBkADKAGiANkAd4A9oB9gH6AP8AikBFgEYgI4AjoBJgCUgFBAKeAVcAsUBaAFpALmAXkAxQBtADcAHOAOoAfYBDoCKgEXgJEASoAnYBQ4CjwFNAKsAWKAtgBcAC5AF2gLzAYMAw0BjwDIwGTgMqAZcAzMBnIDRAGkAOLHgAQCMjgAoAJABHAJsAU2AtwhAyAAWABQADIALgAYgA1ACGAEwAKYAVQAuABiADMAG8APQAjgBSgCxAGEAMoAd4A-wB_gEUAI4ASkAoIBTwCrwFoAWkAuYBigDaAHOAOoAkQBKgCmgFWALFAWUAtEBbAC4AFyALtAZGAycBnIDRCIAIAjICYiAAIAZoCbCUC4ABAACwAKAAZAA4AB-AGAAYgA8ACIAEwAKoAXAAxABmAEMAI4AUYApQBbgDCAGUANkAd8A-wD8AI4AU8Aq8BaAFpALmAYoA3AB1AEOgImARUAi8BIgCjwFigLYAXaAvMBkYDJwGcgNIJgAQCMkgAYAzQE2ALcKQPAAFwAUABUADIAHAAQQAwADKAGgAagA8gCGAIoATAAngBSACqAGIAMwAcwA_ACGAFGAKUAWIAtwBhADIAGUANEAbIA74B9gH6ARYAjEBHAEdAJSAUEAq4BWwC5gF5ANoAbgA-wCHQETAIvASIAk4BOwCrAFigLYAXAAuQBdoC8wGGgMjAZOAy4BnIDSCoAEANpQAMACQAP4BBwE2AKbAW4A.YAAAAAAAAAAA&us_privacy=1---&id=zzz000000000002zzz&ru=https%3A%2F%2Fprebid-server.rubiconproject.com%2Fsetuid%3Fbidder%3D33across%26gdpr%3D1%26gdpr_consent%3DCPtGDMAPtGDMAAGABCENDECkAP_AAG_AAAYgI3td_H_fbX9j-_594ft0eY1f8Pq_ruAzDheNk-8FyJ_W_LwX_2EyNF36pq4KmR4Eu1LBIQNlHEHUDQmQYIkFqTHsak2MpzNKJqJEklMZO2dYGF9vmxFD-YKY705-9vfx2Du9_dv-z5T3r8Xd0AAQACAEAIACAAAAAAAAQAAAAAAAgIAAAAAAAAAAgAQAABAIJIgEGAgIAAACCAAEACAEIAAIABIAAAAAAgAAoAEACBQABKwCIgEAAFAIAIQAAQggIQIBAAABAEgAAAgBYIBAABAIAAQAAAAAACAAEBABYCAAAAgEgAABAACBAQRAAAQhAQEABBACwAAAAEEhgAACACAEAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAIAAAAAgLBIIgACAAFwAUABUADIAHAAPAAgABgADKAGgAagA8gCGAIoATAAngBVADeAHMAPQAfgBCACGAEcAJYATQApQBbgDDAGQAMsAbIA7wB7AD4gH2AfsA_wEAAIpARcBGACNAEcAJMASkAoIBTwCrgFzAMUAawA2gBuAEOgJEATKAnYBQ4CjwFIgKaAWKAtgBcgC8wGDAMNAZOAy4BnIDSAIhhABIAGwASAE0AKWAWcBNgCmwFuAMCDQGAAuACGAGQAMsAbMA-wD8AIAAQUAjABJgCngFXgLQAtIBrADqgIdARMAioBIgCdgFIgLkAZOAzkMACATYApsRAUAEMAMgAZYA2YB9gH4AQAAjABJgCngFXANYAdUBDoCRAE7AKRAXIAyMBk4DORAAIAEgCbBUA4ACgAQwAmABcAEcAMsAjgBV4C0ALSAtgBcgC8wGRgM5FAAQE2DIBoAQwAmACOAGWARwAq4BWwFogLYAXIAvMBkYDORgAEBNg6CmAAuACgAKgAZAA4ACAAF0AMAAygBoAGoAPAAfQBDAEUAJgATwAqgBcADEAGYAN4AcwA9AB-AEMAJYATQAowBSgCxAFvAMIAwwBkADKAGiANkAd4A9oB9gH6AP8AikBFgEYgI4AjoBJgCUgFBAKeAVcAsUBaAFpALmAXkAxQBtADcAHOAOoAfYBDoCKgEXgJEASoAnYBQ4CjwFNAKsAWKAtgBcAC5AF2gLzAYMAw0BjwDIwGTgMqAZcAzMBnIDRAGkAOLHgAQCMjgAoAJABHAJsAU2AtwhAyAAWABQADIALgAYgA1ACGAEwAKYAVQAuABiADMAG8APQAjgBSgCxAGEAMoAd4A-wB_gEUAI4ASkAoIBTwCrwFoAWkAuYBigDaAHOAOoAkQBKgCmgFWALFAWUAtEBbAC4AFyALtAZGAycBnIDRCIAIAjICYiAAIAZoCbCUC4ABAACwAKAAZAA4AB-AGAAYgA8ACIAEwAKoAXAAxABmAEMAI4AUYApQBbgDCAGUANkAd8A-wD8AI4AU8Aq8BaAFpALmAYoA3AB1AEOgImARUAi8BIgCjwFigLYAXaAvMBkYDJwGcgNIJgAQCMkgAYAzQE2ALcKQPAAFwAUABUADIAHAAQQAwADKAGgAagA8gCGAIoATAAngBSACqAGIAMwAcwA_ACGAFGAKUAWIAtwBhADIAGUANEAbIA74B9gH6ARYAjEBHAEdAJSAUEAq4BWwC5gF5ANoAbgA-wCHQETAIvASIAk4BOwCrAFigLYAXAAuQBdoC8wGGgMjAZOAy4BnIDSCoAEANpQAMACQAP4BBwE2AKbAW4A.YAAAAAAAAAAA%26us_privacy%3D1---%26gpp%3D%26gpp_sid%3D%26account%3D%26f%3Db%26uid%3D33XUSERID33X
```

**Comment by bretg** [comment]
> that's geo-located outside those two countries
Thanks for the quick response @ecammit - am US-East cost. I happen to be testing on an Irish site in incognito mode. But from the perspective of your endpoint, I'm US-East.

**Comment by ecammit** [comment]
Two other things:
1. In order to accommodate some portion of GDPR compliance with respect to EU nationals appearing on a US/CA IP address, user state is also maintained through cookies as to whether or not the user was previously found outside the US/CA and retains that state -- which would cause it to not execute to the callback.
2. The callback is executed in a separate request that occurs inside the iframe, not by a redirect of the iframe. I wanted to verify that the expectation was not that the iframe should be redirecting to the callback.

**Comment by ecammit** [comment]
This should be resolved for synacormedia/imds. The issue was caused by the browser blocking an intermediate request to a 3rd party in the chain from lack of proper CORS headers. That 3rd party is no longer in the chain we we switched the request from XHR with credentials to an IMG pixel drop to resolve any future CORS issues (if any).

**Comment by bretg** [comment]
Thanks @ecammit - there's a nice little bump in the graphs already visible in our cluster for /setuid for the synacormedia bidder.

**Comment by vladi-mmg** [comment]
Hi @bretg ,

Thank you for brining this to our attention, I have fixed this issue.

**Comment by ckbo3hrk** [comment]
Hi @bretg 

Thank you for highlighting this problem with adkernel sync. It was a pure configuration issue and it's fixed!

**Comment by bretg** [comment]
Turns out the ucfunnel bidder requires domain registration. Done and confirmed. Submitted pull requests to both repos to add a note in the config about this requirement.

**Comment by bretg** [comment]
@ckbo3hrk - does AdKernel require registration per prebid server domain?

**Comment by ckbo3hrk** [comment]
> @ckbo3hrk - does AdKernel require registration per prebid server domain?

@bretg Nope until now. Is there a way to specify a separate email address in yaml for such purposes?

**Comment by bretg** [comment]
> Is there a way to specify a separate email address in yaml for such purposes?

Yeah, we could just put a comment in the YAML. What's your process @ckbo3hrk ?

**Comment by ckbo3hrk** [comment]
@bretg We've implemented a semi-automatic approach to fix this problem on our side.

**Comment by bretg** [comment]
Thanks to colossus for fixing their macro resolution issue.

**Comment by pm-nilesh-chate** [comment]
Hi @bretg,

Thank you for brining this to our attention, We've fixed this issue.

cc @pm-nitin-shirsat

**Comment by bretg** [comment]
Spoke with ConnectAd. They ignore syncs from VPN which explains what I saw.

**Comment by bretg** [comment]
Spoke with Criteo. They require registration before syncing. This was not clear from their sync documentation, so am working with them on that.

**Comment by bretg** [comment]
33across confirmed that they don't sync in the EEA. So another one that ought to have a note so host companies don't set it up.

**Comment by bretg** [comment]
Lockerdome also confirmed they only operate in the US, so they don't sync in EU

**Comment by bretg** [comment]
Smaato confirmed they don't sync in the GDPR context

**Comment by nickjacob** [comment]
Hi @bretg  -- we've deployed some changes to fix these issues -- when #3022 is merged, our sync issues / that 204 should be resolved
